lldb/source/Target/SwiftLanguageRuntimeDynamicTypeResolution.cpp

```diff
@@ -1142,6 +1142,108 @@ GetTypeFromTypeRef(TypeSystemSwiftTypeRef &ts,
   return ts.RemangleAsType(dem, node);
 }
 
+static llvm::Optional<size_t>
+findFieldWithName(const std::vector<swift::reflection::FieldInfo> &fields,
+                  llvm::StringRef name, std::vector<uint32_t> &child_indexes,
+                  uint32_t offset = 0) {
+  uint32_t index = 0;
+  auto it = std::find_if(fields.begin(), fields.end(), [&](const auto &field) {
+    // A nonnull TypeRef is required for enum cases, where it represents cases
+    // that have a payload. In other types it will be true anyway.
+    if (field.TR == nullptr)
+      return false;
+    if (name != field.Name) {
+      ++index;
+      return false;
+    }
+    return true;
+  });
+  if (it == fields.end())
+    return {};
+  child_indexes.push_back(offset + index);
+  return child_indexes.size();
+}
+
+llvm::Optional<size_t> SwiftLanguageRuntimeImpl::GetIndexOfChildMemberWithName(
+    CompilerType type, llvm::StringRef name, ExecutionContext *exe_ctx,
+    bool omit_empty_base_classes, std::vector<uint32_t> &child_indexes) {
+  auto *ts =
+      llvm::dyn_cast_or_null<TypeSystemSwiftTypeRef>(type.GetTypeSystem());
+  if (!ts)
+    return {};
+
+  using namespace swift::reflection;
+  // Try the static type metadata.
+  const TypeRef *tr = nullptr;
+  auto *ti = GetTypeInfo(type, exe_ctx->GetFramePtr(), &tr);
```

Comment:
this is outside of this patch, but maybe we should return a std::pair or a struct here.

Reply:
I was thinking that too. Initially, only one caller used the `TypeRef`, it was less disruptive to add as an optional out-param. But now more callers are using it, so I will make that follow up change.